site/content/blog/kapp_deploy_oidc_gke.md

```diff
@@ -0,0 +1,119 @@
+---
+title: "kapp deploy on GCP using keyless authentication(OIDC)"
+slug: kapp-deploy-oidc-gke
+date: 2022-07-14
+author: Yash Sethiya
+excerpt: "Use github action OIDC token to authenticate on GCP and deploy using kapp on GKE"
+image: /img/logo.svg
+tags: ['kapp', 'oidc', 'keyless-authentication']
+---
+
+### Who
+
+This article can be helpful for anyone who wants to create the Github Action workflow to authenticate with GCP and how to deploy Kubernetes manifest on GKE using kapp. 
+
+### Why
+
+Traditionally, authenticating from GitHub Actions to Google Cloud required exporting and storing a long-lived JSON service account key, turning an identity management problem into a secrets management problem. 
+
+But now, with GitHub's introduction of OIDC tokens into GitHub Actions Workflows, you can authenticate from GitHub Actions to Google Cloud using OIDC (Workload Identity Federation), removing the need to export a long-lived JSON service account key.
+
+### Benefits
+
+By updating workflows to use OIDC tokens, we can adopt the following good security practices:
+
+- No cloud secrets: No need to duplicate cloud credentials as long-lived GitHub secrets. Instead, you can configure the OIDC trust on GCP, and then update workflows to request a short-lived access token from the cloud provider through OIDC.
+- Authentication and authorization management: Will have more granular control over how workflows can use credentials and also you can control access to cloud resources.
+- Rotating credentials: With OIDC, cloud provider issues a short-lived access token that is only valid for a single job, and then automatically expires.
+
+### How
+
+Now we will see how we can use GitHub Action – [auth](https://github.com/google-github-actions/auth) to set up and configure authentication to Google Cloud. We need to perform the following configurations on GCP - 
+
+1. Create a new Workload Identity Pool (IAM -> Workload Identity Federation -> Workload Identity Pool) and add an OIDC Provider to it with Issuer URL as `https://token.actions.githubusercontent.com`.
+2. Configure the Attribute mapping and conditions of the provider.
+3. Create a service account and connect Workload Identity Pool you just created to the service account by assigned to the Workload Identity User role. For more information, see the [GCP documentation](https://cloud.google.com/iam/docs/workload-identity-federation?_ga=2.114275588.-285296507.1634918453#conditions).
```

Comment:
Create a service account and connect Workload Identity Pool you just created to the service account by assigning it to the Workload Identity User role.

Reply:
Done